src/main/java/com/minecolonies/coremod/inventory/api/AbstractCombinedItemHandler.java

```diff
@@ -0,0 +1,264 @@
+     * is called unexpectedly.
+     *
+     * @param slot  Slot to modify
+     * @param stack ItemStack to set slot to (may be null)
+     * @throws RuntimeException if the handler is called in a way that the handler
+     *                          was not expecting.
+     **/
+    @Override
+    public void setStackInSlot(int slot, final ItemStack stack)
+    {
+        for(IItemHandlerModifiable modifiable : handlers) {
+            if(slot < modifiable.getSlots()) {
+                modifiable.setStackInSlot(slot, stack);
+                return;
+            }
+
+            slot -= modifiable.getSlots();
+        }
+    }
+
+    /**
+     * Returns the number of slots available
+     *
+     * @return The number of slots available
+     **/
+    @Override
+    public int getSlots()
+    {
+        return Arrays.stream(handlers).mapToInt(IItemHandler::getSlots).sum();
+    }
+
+    /**
+     * Returns the ItemStack in a given slot.
+     * <p>
+     * The result's stack size may be greater than the itemstacks max size.
+     * <p>
+     * If the result is null, then the slot is empty.
+     * If the result is not null but the stack size is zero, then it represents
+     * an empty slot that will only accept* a specific itemstack.
+     * <p>
+     * <p/>
+     * IMPORTANT: This ItemStack MUST NOT be modified. This method is not for
+     * altering an inventories contents. Any implementers who are able to detect
+     * modification through this method should throw an exception.
+     * <p/>
+     * SERIOUSLY: DO NOT MODIFY THE RETURNED ITEMSTACK
+     *
+     * @param slot Slot to query
+     * @return ItemStack in given slot. May be null.
+     **/
+    @Override
+    public ItemStack getStackInSlot( int slot)
+    {
+        for(IItemHandlerModifiable modifiable : handlers) {
+            if (slot < modifiable.getSlots()) {
+                return modifiable.getStackInSlot(slot);
+            }
+
+            slot -= modifiable.getSlots();
+        }
+
+        return null;
+    }
+
+    /**
+     * Inserts an ItemStack into the given slot and return the remainder.
+     * The ItemStack should not be modified in this function!
+     * Note: This behaviour is subtly different from IFluidHandlers.fill()
+     *
+     * @param slot     Slot to insert into.
+     * @param stack    ItemStack to insert.
+     * @param simulate If true, the insertion is only simulated
+     * @return The remaining ItemStack that was not inserted (if the entire stack is accepted, then return null).
+     * May be the same as the input ItemStack if unchanged, otherwise a new ItemStack.
+     **/
+    @Override
+    public ItemStack insertItem(int slot, final ItemStack stack, final boolean simulate)
+    {
+        for(IItemHandlerModifiable modifiable : handlers) {
+            if (slot < modifiable.getSlots()) {
+                return modifiable.insertItem(slot, stack, simulate);
+            }
+
+            slot -= modifiable.getSlots();
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Introduce a new variable instead of reusing the parameter "slot". [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://home.kk-sc.de:9000/coding_rules#rule_key=squid%3AS1226)